packages/gatsby/src/internal-plugins/partytown/gatsby-browser.tsx

```diff
@@ -16,9 +18,11 @@ export const onRouteUpdate: GatsbyBrowser[`onRouteUpdate`] = ({
     return
   }
 
-  collectedScriptsByPage.delete(location.pathname)
-
+  // Wait one tick so scripts are collected, then re-initialize Partytown with forwards for the navigated-to page
```

Comment:
have you checked that partytown can be re-initialized without any leaky behavior? Does it only keep 1 sw running? Does it only keep one worker running, etc..

Reply:
Thanks, that's a good callout I didn't check. Added a condition to clean up service workers on CSR.

What's your view on us solving the CSR case in general this way? It reaches into Partytown internals a bit which is dangerous, but does fix broken behavior. For anyone else reading this, https://github.com/BuilderIO/partytown/issues/74 captures the problem

Reply:
Honestly , I would keep it broken until partytown fixes it.

Reply:
Makes sense to me, removed the logic and updated the limitations in docs with links to the relevant discussions/issues in the Partytown repo